experimental/traffic-portal/src/app/api/profile.service.ts

```diff
@@ -71,6 +71,33 @@ export class ProfileService extends APIService {
 		return this.get<Array<ResponseProfile>>(path).toPromise();
 	}
 
+	/**
+	 * Retrieves Profiles associated with a Parameter from the API.
+	 *
+	 * @param p Either a {@link ResponseParameter} or an integral, unique identifier of a Parameter, for which the
+	 * Profiles are to be retrieved.
+	 * @returns The requested Profile(s).
+	 */
+	public async getProfilesByParam(p: number| ResponseParameter): Promise<Array<ResponseProfile>> {
+		let id: number;
+		if (typeof p === "number") {
+			id = p;
+		} else {
+			if (!p.id || p.id < 0) {
+				throw new Error("Parameter id must be defined!");
+			}
+			id = p.id;
+		}
+
+		const path = "profiles";
+		if (id !== undefined) {
+			const params = {param: id};
+			const r = await this.get<Array<ResponseProfile>>(path, undefined, params).toPromise();
+			return r;
+		}
+		return this.get<Array<ResponseProfile>>(path).toPromise();
```

Comment:
The type of `id` down here isn't `undefined` - it's `never`. What that means is that there is no code path that reaches here, so your check doesn't do anything because `id` can never be `undefined` at that point.